main.js

```diff
@@ -0,0 +1,165 @@
+    });
+  }
+};
+
+var setQueryString = function() {
+  var queryString = '';
+  if (searchLocation.zip) {
+    queryString = `zip=${searchLocation.zip}`
+  } else {
+    queryString = `lat=${searchLocation.latitude}&lon=${searchLocation.longitude}`;
+  }
+  return queryString;
+}
+
+var setGoogleString = function(){
+  var googleObj = {};
+  if (searchLocation.zip){
+    googleObj.googleQuery = searchLocation.zip;
+  } else {
+    googleObj.googleQuery = `${searchLocation.latitude}%2C${searchLocation.longitude}`;
+  }
+  return googleObj;
+}
+
+var setConditions = function(weatherObj){
+  var conditions = {};
+  var iconName = weatherObj.weather[0].icon;
+  conditions.temp = Math.round(weatherObj.main.temp);
+  conditions.cityName = weatherObj.name;
+  conditions.weather = weatherObj.weather[0].main;
+  conditions.iconURL = `http://openweathermap.org/img/w/${iconName}.png`,
+  conditions.default = searchLocation.zip === getDefaultLocation().zip //
+  return conditions;
+}
+
+var renderWeatherView = function() {
+  $('#weather-view').empty();
+  var weatherViewHTML = weatherViewTemplate(weatherConditions);
+  var mapHTML = mapTemplate(setGoogleString());
+  $('#weather-view').append(weatherViewHTML, mapHTML);
+
+};
+var renderForecastView = function() {
+  $('#forecast-view').empty();
+  var forecastViewHTML = forecastViewTemplate(forecastData);
+  $('#forecast-view').append(forecastViewHTML);
+}
+
+
+var fetchWeather = function() {
+  var queryString = setQueryString();
+  var url = `http://api.openweathermap.org/data/2.5/weather?${queryString}&units=imperial&appid=6c2fa6a0fe784b355d9a286151f346b6`
+  $.ajax({
+    method: "GET",
+    url: url,
+    dataType: "json",
+    success: function(data) {
+      weatherConditions = setConditions(data);
+      renderWeatherView();
+    },
+
+    error: function(jqXHR, textStatus, errorThrown) {
+      console.log(errorThrown, textStatus)
+    }
+  })
+}
+
+var fetchForecast = function() {
+  var queryString = setQueryString();
+  var url = `http://api.openweathermap.org/data/2.5/forecast?${queryString}&units=imperial&appid=6c2fa6a0fe784b355d9a286151f346b6`
+  $.ajax({
+    method: "GET",
+    url: url,
+    dataType: "json",
+    success: function(data) {
+      //empty forecasts array before adding new data
+      while (forecastData.forecasts.length) {
+        forecastData.forecasts.pop();
+      };
+      var fiveDayArray = data.list.filter(function(forecast, index) {
+        return forecast.dt_txt.indexOf('21:00') !== -1; //gives you forecasts for 9pm UTC/4pm ET/1pm PT
+      })
+      fiveDayArray.forEach(function(day) {
+        var forecastForDay = setConditions(day);
+        forecastForDay.day = moment(day.dt_txt).format('dddd');
+        forecastData.forecasts.push(forecastForDay);
+      })
+      renderForecastView();
+    },
+
+    error: function(jqXHR, textStatus, errorThrown) {
+      console.log(errorThrown, textStatus)
+    }
+  })
+}
+
+getLocation();
```

Comment:
These lines seem pretty hackish (this and 136-139).  Couldn't you have done this better?  Some kind of initialize or setup function would have been better than this.

Reply:
Also, since you put getLocation() here rather than on the handling of the get location button on your interface, its a bad UX.  The user is asked when they first come to the page to share location rather than when requested.  Generally, don't ask for permission until they've done something to ask for permission.